File: athena-boot-pipeline/src/main/java/org/catools/athena/pipeline/common/service/PipelineService.java
package org.catools.athena.pipeline.common.service;


import org.catools.athena.common.service.SaveOrUpdateService;
import org.catools.athena.model.pipeline.PipelineDto;
import org.catools.athena.model.pipeline.PipelineInventoryDto;
import org.catools.athena.model.pipeline.PipelineSummaryDto;
import org.catools.athena.model.pipeline.PipelineTrendPointDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.Instant;
import java.util.List;
import java.util.Optional;

public interface PipelineService extends SaveOrUpdateService<PipelineDto> {

  /**
   * Update pipeline end date
   */
  PipelineDto updatePipelineEndDate(long pipelineId, Instant enddate);

  /**
   * Get pipeline
   */
  Optional<PipelineDto> getPipeline(String pipelineName, String pipelineNumber, String projectCode, String versionCode, String environmentCode);

    Page<PipelineInventoryDto> getAll(
      Pageable pageable,
      String project,
      String version,
      String environment,
      String name,
      String number,
      String state);

    PipelineSummaryDto getSummary(
      String project,
      String version,
      String environment,
      String name,
      String number,
      String state,
      Integer windowDays);

    List<PipelineTrendPointDto> getTrend(
      String project,
      String version,
      String environment,
      String name,
      String number,
      String state,
      Integer windowDays);

}

File: athena-frontend/src/services/pipelineService.js
import api from './api';

const pipelineService = {
  getLastPipeline: (name, number, project, version, environment) => {
    return api.get('/pipeline', {
      params: { name, number, project, version, environment },
    });
  },

  getById: (id) => {
    return api.get(`/pipeline/${id}`);
  },

  getAll: (params) => {
    return api.get('/pipelines', { params });
  },

  create: (pipelineDto) => {
    return api.post('/pipeline', pipelineDto);
  },

  update: (id, pipelineDto) => {
    return api.put(`/pipeline/${id}`, pipelineDto);
  },

  updateEndDate: (pipelineId, endDate) => {
    return api.put('/pipeline', null, {
      params: { pipelineId, date: endDate },
    });
  },

  getExecutions: (pipelineId, params) => {
    return api.get(`/pipeline/${pipelineId}/executions`, { params });
  },

  getMetrics: (pipelineId) => {
    return api.get(`/pipeline/${pipelineId}/metrics`);
  },
};

export default pipelineService;

